lightning/src/util/events.rs

```diff
@@ -360,6 +360,11 @@ pub enum Event {
 		/// If this is `true`, the forwarded HTLC was claimed by our counterparty via an on-chain
 		/// transaction.
 		claim_from_onchain_tx: bool,
+		/// The channel_id of the channel which sent us the payment.  If the channel has been
+		/// force-closed this will be None
+		from_channel_id: Option<[u8; 32]>,
+		/// The channel_id of the channel which we forwarded the payment along
+		to_channel_id: [u8; 32],
```

Comment:
Note, given my earlier comment, these would both need to be an `Option`. You may need a different way to distinguish a force close in the case of `from_channel_id`, though.